kaldb/src/main/java/com/slack/kaldb/clusterManager/ReplicaRestoreService.java

```diff
@@ -0,0 +1,160 @@
+package com.slack.kaldb.clusterManager;
+
+import static com.slack.kaldb.clusterManager.ReplicaCreationService.replicaMetadataFromSnapshotId;
+
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.slack.kaldb.metadata.replica.ReplicaMetadata;
+import com.slack.kaldb.metadata.replica.ReplicaMetadataStore;
+import com.slack.kaldb.metadata.snapshot.SnapshotMetadata;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+import javax.naming.SizeLimitExceededException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Scheduled service responsible for queueing and creating replica snapshots */
```

Comment:
We should expand this definition slightly - currently this doesn't tell us much about the purpose of this class that we can't get from reading the first line of the class definition.

We should say something explaining what this is used for (ie, user requested snapshots that have expired), a mention of the fact it prevents additional snapshots from being created when they already exist, what protections we have in place to prevent a user from overwhelming this service, etc.